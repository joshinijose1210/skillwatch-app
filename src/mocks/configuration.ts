export const ConfigurationMockData = {
    LocationData: {
        pathname: '/',
        state: null,
        search: '',
        hash: '',
        key: ''
    },

    initialState: {
        id: 72,
        organisationId: 1,
        firstName: 'ABC',
        lastName: 'DEF',
        email: 'abc@scalereal.com',
        phoneNumber: '+917060753645',
        employeeId: 'SR0022',
        roleName: 'Manager',
        teamName: 'BA',
        modulePermission: [
            {
                moduleId: 16,
                moduleName: 'Request Feedback',
                view: true,
                edit: true
            },
            {
                moduleId: 3,
                moduleName: 'Feedback',
                view: true,
                edit: true
            },
            {
                moduleId: 10,
                moduleName: 'Performance Review',
                view: true,
                edit: true
            },
            {
                moduleId: 18,
                moduleName: 'Analytics',
                view: true,
                edit: true
            },
            {
                moduleId: 20,
                moduleName: 'Departments',
                view: true,
                edit: true
            },
            {
                moduleId: 4,
                moduleName: 'Teams',
                view: true,
                edit: true
            },
            {
                moduleId: 5,
                moduleName: 'Designations',
                view: true,
                edit: true
            },
            {
                moduleId: 6,
                moduleName: 'Roles & Permissions',
                view: true,
                edit: true
            },
            {
                moduleId: 7,
                moduleName: 'Employees',
                view: true,
                edit: true
            },
            {
                moduleId: 8,
                moduleName: 'KPIs',
                view: true,
                edit: true
            },
            {
                moduleId: 12,
                moduleName: 'Reviews for Team Members',
                view: true,
                edit: true
            },
            {
                moduleId: 13,
                moduleName: 'Check-in with Team Members',
                view: true,
                edit: true
            },
            {
                moduleId: 9,
                moduleName: 'Review Cycles',
                view: true,
                edit: true
            },
            {
                moduleId: 15,
                moduleName: 'Company Information',
                view: true,
                edit: true
            },
            {
                moduleId: 17,
                moduleName: 'Allowed Domains',
                view: true,
                edit: true
            },
            {
                moduleId: 19,
                moduleName: 'Integrations',
                view: true,
                edit: true
            },
            {
                moduleId: 14,
                moduleName: 'User Activity Log',
                view: true,
                edit: true
            }
        ],
        designationName: 'BA - Traineee',
        isLoggedIn: true,
        onboardingFlow: true
    }
};
