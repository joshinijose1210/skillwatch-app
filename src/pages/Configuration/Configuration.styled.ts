import { Button, Link, Text } from '@medly-components/core';
import { SvgIcon } from '@medly-components/icons';
import styled from 'styled-components';
import { StyledTextField, StyledSingleSelect as ParentStyledSingleSelect } from '@common';
import { ScreenWidthWrapper } from '@pages/SelfReview/SelfReviewForm/SelfReviewForm.styled';

export const StyledButtonWrapper = styled('div')`
    display: flex;
    justify-content: flex-end;
    max-width: calc(40% - 5rem);
    gap: 1rem;
`;

export const StyledButton = styled(Button)`
    width: fit-content;
    margin: 2rem 0;
`;

export const StyledScreenWidthWrapper = styled(ScreenWidthWrapper)`
    width: 100%;
    row-gap: 2.5rem;
`;

export const LogoUploadWrapper = styled.div`
    display: flex;
    align-items: center;
    gap: 20rem;
    position: relative;
    justify-content: space-between;
`;
export const StyledInput = styled(StyledTextField)`
    max-width: calc(40% - 5rem);
    padding: 0;
    margin: 0;
`;

export const LogoUpload = styled.div`
    width: calc(40% - 5rem);
`;

export const LogoUploadLabel = styled(Text)`
    margin-bottom: 10px;
    font-weight: 500;
`;

export const LogoNotes = styled.div`
    position: absolute;
    width: 50%;
    right: 0;
    top: 0;
    background-color: ${({ theme }) => theme.colors.grey[50]};
    border-radius: 1rem;
    padding: 2rem;
    color: ${({ theme }) => theme.colors.grey[600]};
    @media (min-width: 1440px) and (max-width: 1536px) {
        width: 47%;
    }
`;

export const LogoNoteHeader = styled.div`
    display: flex;
    align-items: center;
    gap: 1rem;

    ${SvgIcon} {
        width: 2.8rem;
        height: 2.8rem;
    }
`;

export const Notes = styled.ul`
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    ${Text.Style} {
        font-size: 14px;
    }

    ${Link.Style} {
        color: ${({ theme }) => theme.colors.blue[400]};
        text-decoration: underline;

        ${Text.Style} {
            font-size: 14px;
        }
    }
`;

export const StyleContactInput = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;
    width: calc(40% - 5rem);
    div.react-tel-input > input {
        max-height: 45px;
    }
    #medly-textField-input-wrapper {
        width: 100%;
    }
`;

export const StyledErrorText = styled.span`
    position: absolute;
    padding-left: 1.8rem;
    top: 31px;
    color: #d73a43;
    font-size: 1.2rem;
    line-height: 1.6rem;
    margin-top: 1.8rem;
`;

export const SlackContainer = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
`;

export const SlackHeadingContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
`;

export const SlackImage = styled.img`
    height: 20px;
    width: 20px;
    margin: 0 5px;
`;

export const InfoContainer = styled.div`
    display: flex;
    gap: 3rem;
    margin-bottom: 1rem;
`;

export const InfoWrapper = styled.div`
    display: flex;
    flex-direction: column;
    gap: 1rem;
`;
export const StyledRemoveButton = styled(Button)`
    width: fit-content;
    margin: 2rem 0;
    &:not(:disabled):not(:hover),
    &:not(:disabled):not(:active):hover,
    &:not(:disabled):active:hover {
        color: red;
    }
    &:not(:disabled):not(:hover)::after,
    &:not(:disabled):not(:active):hover::after,
    &:not(:disabled):active:hover::after {
        border-color: red;
    }
`;

export const StyledLabel = styled(Text).attrs({
    textVariant: 'body1'
})`
    font-size: ${({ theme }) => theme.contentFontSize};
`;

export const StyledSingleSelect = styled(ParentStyledSingleSelect)`
    width: calc(40% - 5rem);
    max-width: unset;
`;

// commenting this as its not used, might need in future
// export const StyledToggle = styled(Toggle)<{ title: string }>`
//     ${props =>
//         props.title &&
//         `
//             span {
//                 font-size: 1.6rem;
//                 font-weight: 600;
//                 line-height: 1.6rem;
//                 border-bottom: 2px solid blue;
//             }

//             &::after {
//                 display: none;
//             }
//         `}
// `;
