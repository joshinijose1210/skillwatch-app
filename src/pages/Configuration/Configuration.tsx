import { ContactNoWithCode, DropZone, Loader, PageContent } from '@components';
import ListHeader from '@components/reusableComponents/ListHeader';
import { Button, Link, Modal, Text } from '@medly-components/core';
import { EventNoteIcon } from '@medly-components/icons';
import { CancelButton } from '@pages/SelfReview/SelfReviewForm/SelfReviewForm.styled';
import {
    InfoContainer,
    InfoWrapper,
    LogoNoteHeader,
    LogoNotes,
    LogoUpload,
    LogoUploadLabel,
    LogoUploadWrapper,
    Notes,
    StyleContactInput,
    StyledButton,
    StyledInput,
    StyledButtonWrapper,
    StyledScreenWidthWrapper,
    StyledErrorText,
    StyledRemoveButton,
    StyledLabel,
    StyledSingleSelect
} from './Configuration.styled';

import { StyledModalHeader, StyledModalContent, StyledModalActions, StyledModalTitle, StyledTipsTitle } from '@common';

import { useConfiguration } from './useConfiguration';

export const Configuration = () => {
    const {
        companyName,
        hasEditPermission,
        isLoading,
        contactNumber,
        companyLogo,
        handleFileChange,
        handleSaveClick,
        addCompanyInfoLoading,
        organisationId,
        handleCompanyNameChange,
        handleCompanyNumberChange,
        userRoleName,
        errors,
        disableSaveButton,
        data,
        logoUploadError,
        isModalOpen,
        handleRemoveLogo,
        removeLogoLoading,
        companyLogoImg,
        logoStatus,
        isLogoRemoved,
        onCloseModal,
        onOpenModal,
        timeZone,
        allTimeZones,
        setTimeZone
    } = useConfiguration();

    if (isLoading || logoStatus) return <Loader />;
    return (
        <PageContent>
            <ListHeader title="Company" />
            <StyledScreenWidthWrapper>
                <InfoContainer>
                    <InfoWrapper>
                        <StyledLabel>
                            Organisation Id: <Text textWeight="Medium">{organisationId}</Text>
                        </StyledLabel>
                        <StyledLabel>
                            Active Users: <Text textWeight="Medium">{data?.activeUsers}</Text>
                        </StyledLabel>
                        <StyledLabel>
                            Inactive Users: <Text textWeight="Medium">{data?.inactiveUsers}</Text>
                        </StyledLabel>
                    </InfoWrapper>
                </InfoContainer>
                <StyledInput
                    id="COMPANY_NAME"
                    variant="outlined"
                    label="Company Name"
                    fullWidth={true}
                    onChange={handleCompanyNameChange}
                    value={companyName}
                    data-testid="companyNameInput"
                    disabled={userRoleName !== 'Org Admin'}
                    errorText={errors.companyName}
                />
                <StyleContactInput>
                    <ContactNoWithCode
                        inputProps={{
                            name: 'contactNo',
                            autoComplete: 'off'
                        }}
                        country={'in'}
                        componentName={'Configuration'}
                        placeholder="Contact Number"
                        value={contactNumber}
                        onChange={handleCompanyNumberChange}
                        disabled={userRoleName !== 'Org Admin'}
                        isError={errors.contactNumber.length > 0}
                    />
                    <StyledErrorText>{errors.contactNumber}</StyledErrorText>
                </StyleContactInput>
                <StyledSingleSelect
                    options={allTimeZones}
                    label="Timezone"
                    placeholder="Select Company's Timezone"
                    variant="outlined"
                    size="M"
                    onChange={setTimeZone}
                    value={timeZone}
                    isSearchable
                    minWidth="25rem"
                />
                {hasEditPermission && (
                    <>
                        <LogoUploadWrapper>
                            <LogoUpload>
                                <LogoUploadLabel textVariant="h4">Upload Company Logo :</LogoUploadLabel>
                                <DropZone
                                    supportedFiles={['.svg', '.png']}
                                    onDrop={handleFileChange}
                                    file={companyLogo}
                                    accept={{ 'image/svg': ['.svg', '.png'] }}
                                />
                            </LogoUpload>
                            <LogoNotes>
                                <LogoNoteHeader>
                                    <EventNoteIcon iconColor="#000" />
                                    <StyledTipsTitle>Notes:</StyledTipsTitle>
                                </LogoNoteHeader>

                                <Notes>
                                    <Text as="li">The logo should only be in SVG and PNG format.</Text>
                                    <Text as="li">
                                        The logo in PNG or SVG format should be of minimum height of 25px and a minimum width of 25px.
                                    </Text>
                                    <Text as="li">The File size should be maximum 512 KB.</Text>
                                    <Text as="li">Make sure the logo is landscape, and has no whitespace around it.</Text>
                                    <Text as="li">
                                        The logo that will be uploaded, will reflect on the top left corner of the page, throughout the
                                        application, instead of Organisation Name.
                                    </Text>
                                    <Text as="li">
                                        If you have whitespace in the logo, remove it by using{' '}
                                        <Link href="https://svgcrop.com/" target="_blank">
                                            svgcrop tool
                                        </Link>{' '}
                                        for SVG and{' '}
                                        <Link href="https://www.remove.bg/" target="_blank">
                                            remove.bg tool
                                        </Link>{' '}
                                        for PNG format.
                                    </Text>
                                </Notes>
                            </LogoNotes>
                        </LogoUploadWrapper>
                        <StyledButtonWrapper>
                            {!isLogoRemoved && companyLogoImg && companyLogoImg.logoUrl && (
                                <StyledRemoveButton variant="outlined" onClick={onOpenModal} disabled={false} isLoading={false}>
                                    Remove Logo
                                </StyledRemoveButton>
                            )}
                            <StyledButton
                                variant="solid"
                                onClick={handleSaveClick}
                                data-testid="saveButton"
                                disabled={disableSaveButton || logoUploadError.length > 0}
                                isLoading={addCompanyInfoLoading}
                            >
                                Save
                            </StyledButton>
                        </StyledButtonWrapper>
                    </>
                )}
            </StyledScreenWidthWrapper>
            <Modal open={isModalOpen} onCloseModal={onCloseModal}>
                <StyledModalHeader>
                    <StyledModalTitle textVariant="h2">Confirm</StyledModalTitle>
                </StyledModalHeader>
                <StyledModalContent>
                    <Text textVariant="body1" textWeight="Medium">
                        Are you sure you want to remove the logo?
                    </Text>
                </StyledModalContent>
                <StyledModalActions>
                    <CancelButton isLoading={removeLogoLoading} onClick={handleRemoveLogo} variant="outlined">
                        Yes
                    </CancelButton>
                    <Button onClick={onCloseModal} variant="outlined">
                        No
                    </Button>
                </StyledModalActions>
            </Modal>
        </PageContent>
    );
};
