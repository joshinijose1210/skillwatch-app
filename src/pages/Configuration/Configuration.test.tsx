import { RTKRender, cleanup, fireEvent, screen, waitFor } from '@utils/test-utils';
import { Configuration } from './Configuration';
import { Loader } from '@components';
import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import * as reactRouterDom from 'react-router-dom';
import { ConfigurationMockData } from '@mocks/configuration';
const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: jest.fn(),
    useNavigation: mockNavigate
}));
const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');

beforeEach(() => {
    useLocationMock.mockReturnValue(ConfigurationMockData.LocationData);
});

afterEach(() => {
    cleanup();
    mock.reset();
});

describe('Company configuration page', () => {
    it('should take snapshot of loader', () => {
        const { container } = RTKRender(<Configuration />);
        expect(container).toMatchSnapshot();
        const { container: loaderContainer } = RTKRender(<Loader />);
        expect(container.innerHTML).toMatch(loaderContainer.innerHTML);
    });

    it('should take snapshot of loaded page and show current company information without logo', async () => {
        mock.onGet('https://dummy-api.example.com/api/organisation/?id=1').replyOnce(200, {
            id: 1,
            name: 'CompanyOne',
            size: 20,
            contactNo: '+919182848361',
            activeUsers: 6,
            inactiveUsers: 0
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/logo/?id=1').replyOnce(404, {});

        const { container } = RTKRender(<Configuration />, {
            initialState: ConfigurationMockData.initialState
        });

        const name = await screen.findByTestId('companyNameInput');

        expect(container).toMatchSnapshot();

        expect(name).toBeInTheDocument();
        expect(name).toHaveValue('CompanyOne');
    });

    it('should update company name, phonenumber, logo', async () => {
        mock.onGet('https://dummy-api.example.com/api/organisation/?id=1').replyOnce(200, {
            id: 1,
            name: 'CompanyOne',
            size: 20,
            contactNo: '+919182848361',
            activeUsers: 6,
            inactiveUsers: 0
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/logo/?id=1').replyOnce(404, {});
        mock.onPut('https://dummy-api.example.com/api/organisation/1').replyOnce(200, {});
        mock.onGet('https://dummy-api.example.com/api/organisation/logo/?id=1').replyOnce(200, {
            logoUrl: 'https://jadbhasjkasjdbsad.cloudfront.net/1'
        });
        RTKRender(<Configuration />, {
            initialState: ConfigurationMockData.initialState
        });

        const mockLogo = new File([new Blob(['image'])], 'mockLogo.png', { type: 'image/png' });
        const logoUpload = await screen.findByText('Drop File or Upload File here.');
        const companyName = await screen.findByTestId('companyNameInput');
        const companyNumber = await screen.findByPlaceholderText('Contact Number');
        expect(companyName).toBeInTheDocument();
        fireEvent.change(companyName, { target: { value: 'CompanyTwo' } });
        fireEvent.change(companyNumber, { target: { value: '+919876543210' } });
        const save = screen.getByTestId('saveButton');
        const dropLogo = screen.getByTestId('dropLogo');

        fireEvent.click(logoUpload);
        expect(dropLogo).toBeInTheDocument();
        expect(logoUpload).toBeInTheDocument();
        fireEvent.change(dropLogo, { target: { files: [mockLogo] } });
        await waitFor(() => expect(screen.queryByText('Drop File or Upload File here.')).not.toBeInTheDocument());
        expect(save).toBeEnabled();
        expect(await screen.findByText('mockLogo.png')).toBeInTheDocument();
        fireEvent.click(save);
    });

    it('should remove logo when clicked on removelogo button', async () => {
        mock.onGet('https://dummy-api.example.com/api/organisation/?id=1').replyOnce(200, {
            id: 1,
            name: 'CompanyOne',
            size: 20,
            contactNo: '+919182848361',
            activeUsers: 6,
            inactiveUsers: 0
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/logo/?id=1').replyOnce(200, {
            logoUrl: 'https://dummy-url.cloudfront.net/21'
        });
        mock.onPatch('https://dummy-api.example.com/api/organisation/logo/1').replyOnce(200, {});
        RTKRender(<Configuration />, {
            initialState: ConfigurationMockData.initialState
        });
        const resetLogo = await screen.findByText('Remove Logo');
        fireEvent.click(resetLogo);
        expect(await screen.findByText('Are you sure you want to remove the logo?')).toBeInTheDocument();
        fireEvent.click(screen.getByText('No'));
        fireEvent.click(resetLogo);
        expect(await screen.findByText('Are you sure you want to remove the logo?')).toBeInTheDocument();
        fireEvent.click(screen.getByText('Yes'));
        expect(await screen.findByText(/Logo removed successfully/i)).toBeInTheDocument();
    });

    it('should show error when unable to remove logo when clicked on removelogo button', async () => {
        mock.onGet('https://dummy-api.example.com/api/organisation/?id=1').replyOnce(200, {
            id: 1,
            name: 'CompanyOne',
            size: 20,
            contactNo: '+919182848361',
            activeUsers: 6,
            inactiveUsers: 0
        });
        mock.onGet('https://dummy-api.example.com/api/organisation/logo/?id=1').replyOnce(200, {
            logoUrl: 'https://dummy-url.cloudfront.net/21'
        });
        mock.onPatch('https://dummy-api.example.com/api/organisation/logo/1').replyOnce(404, {});
        RTKRender(<Configuration />, {
            initialState: ConfigurationMockData.initialState
        });
        const resetLogo = await screen.findByText('Remove Logo');
        fireEvent.click(resetLogo);
        expect(await screen.findByText('Are you sure you want to remove the logo?')).toBeInTheDocument();
        fireEvent.click(screen.getByText('No'));
        fireEvent.click(resetLogo);
        expect(await screen.findByText('Are you sure you want to remove the logo?')).toBeInTheDocument();
        fireEvent.click(screen.getByText('Yes'));
        expect(await screen.findByText(/Something went wrong./i)).toBeInTheDocument();
    });
});
