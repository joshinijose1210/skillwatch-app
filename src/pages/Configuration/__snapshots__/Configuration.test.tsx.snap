// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company configuration page should take snapshot of loaded page and show current company information without logo 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Company
        </h3>
      </div>
    </div>
    <div
      class="sc-zzdsm5-11 sc-1gx4ncs-2 klTTJe jWuxCF"
    >
      <div
        class="sc-1gx4ncs-15 dvVAZc"
      >
        <div
          class="sc-1gx4ncs-16 gKFYmr"
        >
          <span
            class="sc-1ntgl3l-0 dhgrkE sc-1gx4ncs-18 iFlcGX"
            textalign="initial"
            textcolor="inherit"
            textvariant="body1"
          >
            Organisation Id: 
            <span
              class="sc-1ntgl3l-0 gPMgWC"
              textalign="initial"
              textcolor="inherit"
              textweight="Medium"
            >
              1
            </span>
          </span>
          <span
            class="sc-1ntgl3l-0 dhgrkE sc-1gx4ncs-18 iFlcGX"
            textalign="initial"
            textcolor="inherit"
            textvariant="body1"
          >
            Active Users: 
            <span
              class="sc-1ntgl3l-0 gPMgWC"
              textalign="initial"
              textcolor="inherit"
              textweight="Medium"
            >
              6
            </span>
          </span>
          <span
            class="sc-1ntgl3l-0 dhgrkE sc-1gx4ncs-18 iFlcGX"
            textalign="initial"
            textcolor="inherit"
            textvariant="body1"
          >
            Inactive Users: 
            <span
              class="sc-1ntgl3l-0 gPMgWC"
              textalign="initial"
              textcolor="inherit"
              textweight="Medium"
            >
              0
            </span>
          </span>
        </div>
      </div>
      <div
        class="sc-17d4s39-0 fGqzZg sc-tkiih6-12 sc-1gx4ncs-4 dmAbRN edQmLW"
        id="COMPANY_NAME-input-wrapper"
        minwidth="20rem"
      >
        <div
          class="sc-19px6zp-0 UwvIf"
          disabled=""
          filled="[object Object]"
          fusion="[object Object]"
          minrows="1"
          outlined="[object Object]"
          textvariant="[object Object]"
          variant="outlined"
        >
          <div
            class="sc-lhr232-0 btNOKw"
            variant="outlined"
          >
            <input
              aria-describedby="COMPANY_NAME-helper-text"
              class="sc-hb99wz-0 jaEbhp"
              data-testid="companyNameInput"
              disabled=""
              errortext=""
              id="COMPANY_NAME-input"
              inputsize="M"
              placeholder=" "
              type="text"
              value="CompanyOne"
              variant="outlined"
            />
            <label
              class="sc-1du51u0-0 eMZrST"
              for="COMPANY_NAME-input"
              inputwidth="0"
              variant="outlined"
            >
              Company Name
            </label>
          </div>
        </div>
        
      </div>
      <div
        class="sc-1gx4ncs-10 fJiBRO"
      >
        <div
          class="sc-1fl0hxw-0 aTIXC"
          component="Configuration"
          disabled=""
        >
          <div
            class=" react-tel-input "
          >
            <div
              class="special-label"
            >
              Phone
            </div>
            <input
              class="form-control "
              disabled=""
              placeholder="Contact Number"
              type="tel"
              value="+91 91828-48361"
            />
            <div
              class="flag-dropdown "
            >
              <div
                aria-haspopup="listbox"
                class="selected-flag"
                role="button"
                tabindex="0"
                title="India: + 91"
              >
                <div
                  class="flag in"
                >
                  <div
                    class="arrow"
                  />
                </div>
              </div>
            </div>
          </div>
           
        </div>
        <span
          class="sc-1gx4ncs-11 jXuONC"
        />
      </div>
      <div
        class="sc-1hcsztp-0 jtzMlh sc-tkiih6-13 sc-1gx4ncs-19 cecHDR iJZdmz"
        id="timezone-wrapper"
        minwidth="25rem"
        variant="outlined"
      >
        <div
          class="sc-17d4s39-0 UfceN"
          id="timezone-input-wrapper"
          minwidth="25rem"
        >
          <div
            class="sc-19px6zp-0 BrSGZ"
            filled="[object Object]"
            fusion="[object Object]"
            minrows="1"
            outlined="[object Object]"
            textvariant="[object Object]"
            variant="outlined"
          >
            <div
              class="sc-lhr232-0 btNOKw"
              variant="outlined"
            >
              <input
                aria-describedby="timezone-helper-text"
                autocomplete="off"
                class="sc-hb99wz-0 jaEbhp"
                errortext=""
                id="timezone-input"
                inputsize="M"
                placeholder="Select Company's Timezone"
                type="text"
                value=""
                variant="outlined"
              />
              <label
                class="sc-1du51u0-0 eMZrST"
                for="timezone-input"
                inputwidth="0"
                variant="outlined"
              >
                Timezone
              </label>
            </div>
            <svg
              class="sc-1pf3x8a-0 klDMsE sc-dynlvq-0 bdmviR"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                fill="#000"
              />
            </svg>
          </div>
          
        </div>
      </div>
      <div
        class="sc-1gx4ncs-3 fAmxyw"
      >
        <div
          class="sc-1gx4ncs-5 jtbaWJ"
        >
          <h4
            class="sc-1ntgl3l-0 iEGCWU sc-1gx4ncs-6 dObUVQ"
            textalign="initial"
            textcolor="inherit"
            textvariant="h4"
          >
            Upload Company Logo :
          </h4>
          <div
            class="sc-14diotp-4 kELNNF"
          >
            <div
              class="sc-14diotp-3 eyQRCT"
              role="presentation"
              tabindex="0"
            >
              <img
                alt="upload"
                class="sc-14diotp-6 cIgHdk"
                src="[object Object]"
              />
              <label
                class="sc-14diotp-2 bgzcdi"
                id="button"
              >
                 Drop File or Upload File here.
              </label>
              <input
                accept="image/svg,.svg,.png"
                class="sc-14diotp-1 jZvqTL"
                data-testid="dropLogo"
                multiple=""
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
                tabindex="-1"
                type="file"
              />
            </div>
            
            <em
              class="sc-14diotp-7"
            >
              (Only .svg and .png file will be accepted)
            </em>
          </div>
        </div>
        <div
          class="sc-1gx4ncs-7 GaCcE"
        >
          <div
            class="sc-1gx4ncs-8 gOZXTk"
          >
            <svg
              class="sc-1pf3x8a-0 fYFAcU"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M18 4h1c1.1 0 2 .9 2 2v14c0 1.1-.9 2-2 2H5a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h1V3c0-.55.45-1 1-1s1 .45 1 1v1h8V3c0-.55.45-1 1-1s1 .45 1 1v1zM8 11h8c.55 0 1 .45 1 1s-.45 1-1 1H8c-.55 0-1-.45-1-1s.45-1 1-1zm10 9H6c-.55 0-1-.45-1-1V9h14v10c0 .55-.45 1-1 1zM8 15h5c.55 0 1 .45 1 1s-.45 1-1 1H8c-.55 0-1-.45-1-1s.45-1 1-1z"
                fill="#000"
                fill-opacity=".54"
                fill-rule="evenodd"
              />
            </svg>
            <h3
              class="sc-1ntgl3l-0 lhJlYH sc-tkiih6-22 isRtxH"
              textalign="initial"
              textcolor="inherit"
              textvariant="h3"
              textweight="Strong"
            >
              Notes:
            </h3>
          </div>
          <ul
            class="sc-1gx4ncs-9 jaguPm"
          >
            <li
              class="sc-1ntgl3l-0 iDGDWd"
              textalign="initial"
              textcolor="inherit"
            >
              The logo should only be in SVG and PNG format.
            </li>
            <li
              class="sc-1ntgl3l-0 iDGDWd"
              textalign="initial"
              textcolor="inherit"
            >
              The logo in PNG or SVG format should be of minimum height of 25px and a minimum width of 25px.
            </li>
            <li
              class="sc-1ntgl3l-0 iDGDWd"
              textalign="initial"
              textcolor="inherit"
            >
              The File size should be maximum 512 KB.
            </li>
            <li
              class="sc-1ntgl3l-0 iDGDWd"
              textalign="initial"
              textcolor="inherit"
            >
              Make sure the logo is landscape, and has no whitespace around it.
            </li>
            <li
              class="sc-1ntgl3l-0 iDGDWd"
              textalign="initial"
              textcolor="inherit"
            >
              The logo that will be uploaded, will reflect on the top left corner of the page, throughout the application, instead of Organisation Name.
            </li>
            <li
              class="sc-1ntgl3l-0 iDGDWd"
              textalign="initial"
              textcolor="inherit"
            >
              If you have whitespace in the logo, remove it by using
               
              <a
                class="sc-iayd5c-0 hssINh"
                href="https://svgcrop.com/"
                target="_blank"
              >
                <span
                  class="sc-1ntgl3l-0 hGagSU"
                  textalign="initial"
                  textcolor="inherit"
                >
                  svgcrop tool
                </span>
              </a>
               
              for SVG and
               
              <a
                class="sc-iayd5c-0 hssINh"
                href="https://www.remove.bg/"
                target="_blank"
              >
                <span
                  class="sc-1ntgl3l-0 hGagSU"
                  textalign="initial"
                  textcolor="inherit"
                >
                  remove.bg tool
                </span>
              </a>
               
              for PNG format.
            </li>
          </ul>
        </div>
      </div>
      <div
        class="sc-1gx4ncs-0 jypDwj"
      >
        <button
          class="sc-1k2druj-0 eLgfbc sc-1gx4ncs-1 kMvWqq"
          data-testid="saveButton"
          disabled=""
          edges="square"
          type="button"
          variant="solid"
        >
          <span
            class="sc-1ntgl3l-0 eOsHHD"
            textalign="initial"
            textcolor="inherit"
            textvariant="button1"
          >
            Save
          </span>
        </button>
      </div>
    </div>
  </main>
</div>
`;

exports[`Company configuration page should take snapshot of loader 1`] = `
<div>
  <div
    class="sc-mbkfkf-0 KnoHL"
  >
    <main
      class="sc-1ehavxz-0 dDyhBJ"
    >
      <div
        class="sc-6d0kjn-0 dkGZbP"
      >
        <svg
          class="sc-433e5x-0 dVjAFq"
          height="1em"
          viewBox="0 0 100 100"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.6 3.5C5.9 13.6-6.6 42.7 3.5 68.4c10.1 25.7 39.2 38.3 64.9 28.1l-3.1-7.9C44 97 19.9 86.6 11.5 65.3c-8.4-21.3 2-45.4 23.3-53.8l-3.2-8z"
          >
            <animatetransform
              attributeName="transform"
              attributeType="XML"
              dur="2s"
              from="0 50 50"
              repeatCount="indefinite"
              to="360 50 50"
              type="rotate"
            />
          </path>
          <path
            d="M42.3 39.6c5.7-4.3 13.9-3.1 18.1 2.7 4.3 5.7 3.1 13.9-2.7 18.1l4.1 5.5c8.8-6.5 10.6-19 4.1-27.7-6.5-8.8-19-10.6-27.7-4.1l4.1 5.5z"
          >
            <animatetransform
              attributeName="transform"
              attributeType="XML"
              dur="1s"
              from="0 50 50"
              repeatCount="indefinite"
              to="-360 50 50"
              type="rotate"
            />
          </path>
          <path
            d="M82 35.7C74.1 18 53.4 10.1 35.7 18S10.1 46.6 18 64.3l7.6-3.4c-6-13.5 0-29.3 13.5-35.3s29.3 0 35.3 13.5l7.6-3.4z"
          >
            <animatetransform
              attributeName="transform"
              attributeType="XML"
              dur="2s"
              from="0 50 50"
              repeatCount="indefinite"
              to="360 50 50"
              type="rotate"
            />
          </path>
        </svg>
      </div>
    </main>
  </div>
</div>
`;
